lib/src/time_only.dart

```diff
@@ -0,0 +1,75 @@
+part of '../kiota_abstractions.dart';
+
+/// Interface for a time only object that represents a time of day.
+///
+/// This interface provides an abstraction layer over time only objects.
+/// It is used to represent time only values in a serialization format agnostic
+/// way.
+abstract class TimeOnly {
+  /// Extracts the time part of a [DateTime] and creates an object implementing
+  /// [TimeOnly].
+  factory TimeOnly.fromDateTime(DateTime dateTime) {
+    return _TimeOnlyImpl(
+      hours: dateTime.hour,
+      minutes: dateTime.minute,
+      seconds: dateTime.second,
+      milliseconds: dateTime.millisecond,
+    );
+  }
+
+  /// This factory uses the [DateTime.parse] method to create an object
+  /// implementing [TimeOnly].
+  factory TimeOnly.fromDateTimeString(String dateTimeString) {
+    final dateTime = DateTime.parse('2024-01-01 $dateTimeString');
```

Comment:
Not sure if this is the best way to achieve "time only" parsing. Should I just implement this parser myself?

Reply:
Ok, I'll keep it the way it is for now.